hackathon_site/event/serializers.py

```diff
@@ -10,18 +10,15 @@ class Meta:
         fields = ("id", "name")
 
 
-class TeamSerializer(serializers.ModelSerializer):
+class UserInSerializer(serializers.ModelSerializer):
     class Meta:
-        model = Team
-        fields = (
-            "id",
-            "team_code",
-            "created_at",
-            "updated_at",
-        )
+        model = User
+        fields = ("id", "first_name", "last_name", "email")
 
 
 class ProfileSerializer(serializers.ModelSerializer):
+    user = UserInSerializer(read_only=True)
```

Comment:
Gotta be careful with this - the `UserSerializer` uses `ProfileSerializer`, so you'll end up with infinite recursion I think. You may need to define separate `ProfileInUserSerializer` and `ProfileInTeamSerializer`s

Reply:
yeah that's true. But to my surprise, it worked. I am afraid there will be some potential issues appear in the future if wrote like this